source/common/http/http1/codec_impl.cc

```diff
@@ -411,6 +411,11 @@ int ServerConnectionImpl::onHeadersComplete(HeaderMapImplPtr&& headers) {
         http_parser_pause(&parser_, 1);
       }
 
+    } else if (headers->Method()->value() == "POST" || headers->Method()->value() == "PUT") {
```

Comment:
To avoid the string compares here we can pull the already parsed method out of the http_parser struct.